bin/i18n/resources/dashboard/course_content/sync_in.rb

```diff
@@ -342,7 +342,7 @@ def process
               # We want to make sure to categorize HoC scripts as HoC scripts even if
               # they have a version year, so this ordering is important
               script_i18n_directory =
-                if Unit.unit_in_category?('hoc', script.name)
+                if ScriptConstants.unit_in_category?('hoc', script.name)
```

Comment:
Shouldn't we use `Unit.in_initiative?('HOC')` instead?
https://github.com/code-dot-org/code-dot-org/pull/58949/files#diff-debae58161437b537cd58e1c7ab8fc3ddda361cbbe871088a127b2592e4429d7L832-R834

Reply:
I would love to get all uses of `unit_in_category?` moved over to `in_initiative?`. However I am not sure I understand the environment in which the i18n sync runs. Does it have all units seeded into its database? if so, the next thing to check is whether all units in the hoc category are also properly marked as in the HOC initiative. I can try that and get back to you with an update.

Reply:
mi = marketing initiative
sc = script constants
here are the discrepancies I see between the two data sources:
```
[development] dashboard > mi_offerings = CourseOffering.where(marketing_initiative: 'HOC').map(&:key).compact.sort

[development] dashboard > sc_names = ScriptConstants::CATEGORIES[:hoc].compact
[development] dashboard > sc_offerings = Unit.where(name: sc_names).map(&:get_course_version).map(&:course_offering).map(&:key).compact.sort

[development] dashboard > mi_offerings - sc_offerings
=> ["pixelation"]
[development] dashboard > sc_offerings - mi_offerings
=> ["ai-ethics", "hour-of-code", "how-ai-works"]
```

Reply:
If you want to depend on marketing initiative, you will be tied to the field that is used by Acquisition Products to control what shows up on the studio.code.org/catalog page. this could be good, if you don't want to miss new HOC tutorials as they are added. but it could be bad if you want more control. @artem-vavilov @mgc1194 let me know if you think you'd like to start using marketing initiative and I'll be happy to help. I would suggest a separate PR for this.

Reply:
sorry to go on but here is the difference with unit names instead of course offering names, since it looks like those might be what you use in the i18n code:
```
[development] dashboard > mi_names = CourseOffering.where(key: mi_offerings).all.map(&:course_versions).flatten.map(&:content_root).map(&:name).compact.sort

[development] dashboard > sc_names - mi_names
=> ["Hour of Code", "ai-ethics-2023", "generation-ai", "how-ai-works-2023"]
[development] dashboard > mi_names - sc_names
=> ["pixelation"]
```

Reply:
We only need the category/initiative to determine the files' directory before uploading them to Crowdin. The possible folders are `Hour of Code`, `other`, or `year`, giving additional context to translators and helping prioritize some files over others. All translatable scripts are still being translated.
However, we prevent uploading duplicated files when they appear in multiple directories, therefore changing the initiative of a Unit will prevent from that unit to sync i18n content unless files are manually moved in the i18n directory.

Reply:
It sounds like the teacher tools preference is using the `in_initiative?` method, so I will go ahead and use that method instead.

Reply:
> It sounds like the teacher tools preference is using the in_initiative? method, so I will go ahead and use that method instead.

yes, sounds great! the more calls to `unit_in_category?` we can eliminate from i18n code the better. This will get us one step closer to removing the big chunks of script_constants which are duplicate state from whats in course_offerings.marketing_initiative.

Reply:
The only call left to `unit_in_category?` I could find is in [`certificate_image.rb`](https://github.com/code-dot-org/code-dot-org/blob/243954975c96d28e16e2e6704aa419ac6208c7ef/dashboard/lib/certificate_image.rb#L331)

There are no more calls to `unit_in_category?` in the i18n side of things.

Reply:
Sorry I misspoke -- there are other dependencies from i18n code on the following parts of ScriptConstants:
* https://github.com/code-dot-org/code-dot-org/blob/5259a94ee7b7c67a537cc8efa64b347cf2105689/dashboard/lib/script_constants.rb#L375-L380
* https://github.com/code-dot-org/code-dot-org/blob/5259a94ee7b7c67a537cc8efa64b347cf2105689/dashboard/lib/script_constants.rb#L310-L332

The fix for these would be different, possibly one of:
1. write special logic for translatability, e.g. "course is CSF, CSD or HOC and is in preview or published state", so that we no longer explicitly have to set when a course is ready to be translated
2. add a new flag to levelbuilder, figure out who (non-engineering) should mark each course / unit as translatable when it is time
3. do nothing, leave the code how it is

since teacher tools appears to make the updates to the list of which units are translatable, I will consult with the rest of the teacher tools team on which option would be best.